app/assets/javascripts/react/decorators/resourcesbox/ActionsDropdown.cjsx

```diff
@@ -0,0 +1,154 @@
+          createHoverActionItem(
+            if f.present(batchPermissionEditables) then f.curry(callbacks.onBatchPermissionsEdit)(batchPermissionEditables),
+            'media_entries_permissions',
+            batchPermissionEditables.length,
+            'lock',
+            'resources_box_batch_actions_managepermissions')}
+
+        {if showActions.managePermissionsSets and false
+          batchPermissionSetsEditables = SelectionScope.batchPermissionResources(selection, ['Collection']) if selection
+          createHoverActionItem(
+            if f.present(batchPermissionSetsEditables) then f.curry(callbacks.onBatchPermissionsSetsEdit)(batchPermissionSetsEditables),
+            'collections_permissions',
+            batchPermissionSetsEditables.length,
+            'lock',
+            'resources_box_batch_actions_sets_managepermissions')}
+
+        {if showActions.save
+          <MenuItem className="separator"/>}
+        {if showActions.save
+          <MenuItem onClick={(if f.present(config.filter) then f.curry(callbacks.onCreateFilterSet)(config))}>
+            <Icon i="filter" mods="ui-drop-icon"/> {t('resources_box_batch_actions_save')}
+          </MenuItem>}
+
+      </Dropdown.Menu>
+    </Dropdown>
+
+  actionsDropdown
+
+
+
+highlightingRules = (item, isSelected) ->
+
+  highlighting_rules = [
+    {
+      hoverMenuId: 'media_entries_edit'
+      rule: () -> (!SelectionScope.batchMetaDataResource(item.serialize()) or
+        item.type != 'MediaEntry' or not isSelected)
+    }
+    {
+      hoverMenuId: 'collections_edit'
+      rule: () -> (!SelectionScope.batchMetaDataResource(item.serialize()) or
+        item.type != 'Collection' or not isSelected)
+    }
+    {
+      hoverMenuId: 'media_entries_permissions'
+      rule: () -> (!SelectionScope.batchPermissionResource(item.serialize()) or
+        item.type != 'MediaEntry' or not isSelected)
+    }
+    {
+      hoverMenuId: 'collections_permissions'
+      rule: () -> (!SelectionScope.batchPermissionResource(item.serialize()) or
+        item.type != 'Collection' or not isSelected)
+    }
+    {
+      hoverMenuId: 'add_to_set'
+      rule: () ->
+        (!SelectionScope.batchPermissionResource(item.serialize()) or
+          (item.type != 'MediaEntry' and item.type != 'Collection') or not isSelected)
+    }
+    {
+      hoverMenuId: 'remove_from_set'
+      rule: () -> (!SelectionScope.batchPermissionResource(item.serialize()) or
+        (item.type != 'MediaEntry' and item.type != 'Collection') or not isSelected)
+    }
+  ]
+
+  highlighting_rules
+
+
+isResourceNotInScope = (item, isSelected, hoverMenuId) ->
+  found_rules = f.filter(highlightingRules(item, isSelected), {hoverMenuId: hoverMenuId})
+  not f.isEmpty(found_rules) and f.first(found_rules).rule() == true
+
+
+
+ActionsDropdown = {
+  createActionsDropdown: createActionsDropdown
+  isResourceNotInScope: isResourceNotInScope
+}
+
+
+module.exports = ActionsDropdown
```

Comment:
Sehr gut rausextrahiert. Da sind ganz schön viele Business-rules drin, viel dynamischer wird es wahrscheinlich auch in Zukunft nicht werden. D.h. wenn in bestimmten Views neue Actions hinzukommen werden die auch so übergeben wie und das Menü muss sie alle handlen können.